src/test/scala/linting/rule/LintConflictingModuleNamesSpec.scala

```diff
@@ -0,0 +1,136 @@
+// See LICENSE for license details.
+
+package freechips.rocketchip.linting.rule
+
+import firrtl._
+import firrtl.ir.Port
+import firrtl.annotations._
+import firrtl.testutils.{FirrtlMatchers, FirrtlPropSpec}
+
+case class StableNameAnnotation(target: IsModule) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): StableNameAnnotation = this.copy(target = newTarget)
+}
+
+case class UnstableNameAnnotation(target: IsModule) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): UnstableNameAnnotation = this.copy(target = newTarget)
+}
+
+class RenameDesiredNamesSpec extends FirrtlPropSpec with FirrtlMatchers {
+  val transform = new RenameDesiredNames
+
+  case class TestCase(
+    input: String,
+    annos: Seq[Annotation]
+  )
+
+  def stabilizeNames(testCase: TestCase): CircuitState = {
+    stabilizeNames(CircuitState(Parser.parse(testCase.input), UnknownForm, testCase.annos))
+  }
+
+  def stabilizeNames(state: CircuitState): CircuitState = {
+    transform.runTransform(state)
+  }
+
+  def test(testCases: TestCase *): Unit = {
+    val firstState = stabilizeNames(testCases.head)
+    val firstUnstableNames = firstState.annotations.collect {
+      case a: UnstableNameAnnotation => a
+    }
+    val firstStableNames = firstState.annotations.collect {
+      case a: StableNameAnnotation => a
+    }
+
+    testCases.tail.foldLeft(firstStableNames, firstUnstableNames) {
+      case ((stable, unstable), testCase) =>
+        val state = stabilizeNames(testCase)
+        val currUnstableNames = state.annotations.collect {
+          case a: UnstableNameAnnotation => a
+        }
+        val currStableNames = state.annotations.collect {
+          case a: StableNameAnnotation => a
+        }
+
+        currStableNames should be (stable)
+        currUnstableNames should not be (unstable)
+        (currStableNames, currUnstableNames)
+    }
+  }
+
+  property("It should rename modules to desired names if only one module") {
+    val top = CircuitTarget("Foo")
+    test(
+      TestCase(
+        """|circuit Foo:
+           |  module Bar_1:
+           |    output in1: UInt<1>
+           |  module Bar_2:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |  module Bar_3:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |    output in3: UInt<1>
+           |  module Foo:
+           |    inst bar_1 of Bar_1
+           |    inst bar_2 of Bar_2
+           |    inst bar_3 of Bar_3
+           |""".stripMargin,
+        Seq(
+          UnstableNameAnnotation(top.module("Bar_1")),
+          StableNameAnnotation(top.module("Bar_2")),
+          StableNameAnnotation(top.module("Bar_3")),
```

Comment:
I don't understand what the UnstableName/StableName annotations are doing in the test

Reply:
stable/unstable are supposed to specify that a name changes/does not change from one run to the next.